### PR TITLE
[FIX] Let decoder accept list of paths as input

### LINKED ISSUE #4226: [BUG] Backward incompatible change in Decoder
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [X] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

- Linux Ubuntu 22.04

### Python version

- [ ] 3.12
- [ ] 3.11
- [X] 3.10
- [ ] 3.9
- [ ] 3.8

### nilearn version

0.11.dev

### Expected behavior

The doc build on main is failing because merge of #4188 broke one example. `decoder.decision_function` is called on a list of file paths which is not acceptable input. This input is neither documented nor tested on any of the decoder methods unless I missed something

### Current behavior & error messages

This is what I got:

Calling `decoder.predict` with a list of filepaths will pass this input to `decoder.decision_function` and fail

### Steps and code to reproduce bug

See https://nilearn.github.io/dev/auto_examples/02_decoding/plot_oasis_vbm.html#sphx-glr-auto-examples-02-decoding-plot-oasis-vbm-py

### PR DESCRIPTION
**Pull request by ymzayek**
- Closes #4226

### CODE REVIEW

File: nilearn/decoding/tests/test_decoder.py [hunk 1]
```diff
@@ -1037,3 +1038,35 @@ def test_decoder_decision_function(binary_classification_data):
     X = model.masker_.transform(X)
     assert X.shape[1] == model.coef_.shape[1]
     model.decision_function(X)
+
+
+def test_decoder_strings_filepaths_input(
+    tiny_binary_classification_data, tmp_path
+):
+    """Smoke test for decoder methods to accept list of paths as input."""
```
Comment (Remi-Gau) [inline nilearn/decoding/tests/test_decoder.py:1046]: should we add a link to the issue?

Comment (ymzayek) [reply nilearn/decoding/tests/test_decoder.py:1046]: sure can do

**Review by bthirion — COMMENTED** [review]
We probably also have to adapt Decoder heuristics with the percentile selection ? What about clustering ?

**Comment by ymzayek** [comment]
> We probably also have to adapt Decoder heuristics with the percentile selection ? What about clustering ?

I'm not sure I understand. This PR is just fixing a regression that I caused with #4188. It's really just about the input of filepaths as lists of path-like or string to be allowed again

**Comment by ymzayek** [comment]
Yes I see, you're comments are relevant o #4205 (I think the naming of the PRs might have caused a mix up!). I will copy your comment to that PR.

This PR should be good to go. It's not related to any of the surface stuff :)

File: nilearn/decoding/decoder.py [hunk 2]
```diff
@@ -861,18 +862,20 @@ def decision_function(self, X):
 
         Parameters
         ----------
-        X: list of Niimg-like objects
+        X: Niimg-like, :obj:`list` of Niimg-like objects, or \
+           :obj:`list` of :obj:`str` or path-like
```
Comment (Remi-Gau) [inline nilearn/decoding/decoder.py:866]:
```suggestion
        X: Niimg-like, :obj:`list` of either \
           Niimg-like objects or :obj:`str` or path-like
```

File: nilearn/decoding/decoder.py [hunk 3]
```diff
@@ -893,7 +896,8 @@ def predict(self, X):
 
         Parameters
         ----------
-        X: Niimg-like
+        X: Niimg-like, :obj:`list` of Niimg-like objects, or \
+            :obj:`list` of :obj:`str` or path-like
```
Comment (Remi-Gau) [inline nilearn/decoding/decoder.py:900]:
```suggestion
        X: Niimg-like, :obj:`list` of either \
           Niimg-like objects or :obj:`str` or path-like
```

**Review by Remi-Gau — APPROVED** [review]
Good with me.

Feel free to ignore the doc string suggestions if that does not feel cleared to you.

**Review by bthirion — APPROVED** [review]
LGTM, thx

**Pull request merged**